lib/optimizer/src/Transformer/ServerSideRendering.php

```diff
@@ -620,4 +713,267 @@ private function isAmpExperimentUsed(DOMElement $element)
+     * @param string     $mediaQueryStyle CSS template for a media query style.
+     * @return string Extract custom CSS styling.
+     */
+    private function extractAttributeCss(Document $document, DOMElement $element, $attributeName, $attributeValue, $mainStyle, $mediaQueryStyle)
+    {
+        if (empty($attributeValue)) {
+            return '';
+        }
+
+        $sourceSizes = explode(',', $attributeValue);
+        $lastItem    = trim(array_pop($sourceSizes), self::CSS_TRIM_CHARACTERS);
+
+        if (empty($lastItem)) {
+            throw InvalidHtmlAttribute::fromAttribute($attributeName, $element);
+        }
+
+        $customCss = sprintf($mainStyle, $lastItem);
+
+        foreach ($sourceSizes as $sourceSize) {
+            $matches = [];
+            if (preg_match(self::CSS_DIMENSION_WITH_MEDIA_CONDITION_REGEX_PATTERN, $sourceSize, $matches)) {
+                $mediaCondition = trim($matches['media_condition'], self::CSS_TRIM_CHARACTERS);
+
+                if (empty($mediaCondition)) {
+                    throw InvalidHtmlAttribute::fromAttribute($attributeName, $element);
+                }
+
+                $dimension = trim($matches['dimension'], self::CSS_TRIM_CHARACTERS);
+
+                if (empty($dimension)) {
+                    throw InvalidHtmlAttribute::fromAttribute($attributeName, $element);
+                }
+
+                $customCss .= sprintf($mediaQueryStyle, $mediaCondition, $dimension);
+            }
+        }
+
+        $elementId = $document->getElementId($element);
+        $customCss = str_replace('__ID__', $elementId, $customCss);
+
+        return $customCss;
+    }
+
+    /**
+     * Extract the custom CSS styling from a 'media' attribute.
+     *
+     * __media__
+     * "The value of media is a media query. If the query does not match, the element is not rendered and its resources
+     * and potentially its child resources will not be fetched. If the browser window changes size or orientation, the
+     * media queries are re-evaluated and elements are hidden and shown based on the new results."
+     *
+     * @param Document   $document       Document containing the element to adapt.
+     * @param DOMElement $element        Element to adapt.
+     * @param string     $attributeName  Name of the attribute to be extracted.
+     * @param string     $attributeValue Value of the attribute to extract the CSS styling from.
+     * @return string Extract custom CSS styling.
+     */
+    private function extractMediaAttributeCss(Document $document, DOMElement $element, $attributeName, $attributeValue)
+    {
+        $attributeValue = trim($attributeValue, self::CSS_TRIM_CHARACTERS);
+
+        if (empty($attributeValue)) {
+            return '';
+        }
+
+        if ($attributeValue[0] === '(') {
+            $attributeValue = 'all and ' . $attributeValue;
+        }
+
+        return sprintf(
+            '@media not %s{#%s{display:none}}',
```

Comment:
What about `not screen and (min-width: 650px)` as input? Won't this result in `not not screen and (min-width: 650px)`?

Reply:
Another thing: would be nice to combine multiple rules with the same media query, e.g.:

```
@media (min-width: 650px){
    #amp_o_4,#amp_o_5,#amp_o_6{
        display:none
    }
}
```

Reply:
Regarding the `not not` case, I did have logic to prevent that in a previous iteration, but I think that got lost in the latest rewrite. I'll have to add that back in. Good catch!

Reply:
Wouldn't it make more sense to do the rule combining in the minified?

Reply:
You're right. CSS optimizer would be the right place. In this particular case it's easy to merge the rules so I did it in the transform, but for example merging the sizes/heights rules would be way more complex to implement hence I didn't implement it either.

Reply:
In the case of the AMP plugin at least at present, the CSS minification runs earlier before the Optimizer. This is because the CSS needs to be minified at a point where we can still strip out CSS that causes it to go over the budget. So I'd recommend outputting pre-minified CSS in the PHP Optimizer and not rely on a minification transformer which would add quite a bit more processing time.

Reply:
The wrong negation was fixed via https://github.com/ampproject/amp-wp/pull/4799